### ISSUE TITLE
Deterministic sampling for PGPolicy

### ISSUE THREAD

**Issue by ultmaster** [thu-ml/tianshou#353]
- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [x] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [x] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, torch, numpy, sys
  print(tianshou.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
  ```

```
0.4.1 1.8.0 1.19.2 3.8.8 (default, Feb 24 2021, 21:46:12) 
[GCC 7.3.0] linux
```

I was wondering why tianshou's PGPolicy, A2CPolicy and PPOPolicy always use `dist.sample()`. This causes undeterminstic behavior. On the other hand, many other RL frameworks (e.g., rllib) provides ways to make it determinstic by using argmax (https://github.com/ray-project/ray/blob/cecfc3b43b7a1a7ec4398645556ef64c21704ea8/rllib/models/torch/torch_action_dist.py#L362).

It would be nice if tianshou can provide an init parameter that enables deterministic sampling in case of evaluation mode, e.g.,

```python
if self.training:
  act = self.dist_fn(logits).sample()
else:
  act = logits.argmax()
```

or maybe can we make the `dist_fn` aware of current mode?

**Comment by ChenDRAG** [comment]
Will consider supporting it soon. Thank you very much for your suggestion. Meanwhile, you can try inherite/rewrite offically provided Policy to customize deterministic sampling. A pr to fix this is also very welcome.
(However, please note that in onpolicy algorithms, `std` of action distribution generated by actor is usually a trainable Parameter, thus if properly trained, maybe they will be of very small value and don't actually matter when the algorithm converges?)

**Comment by Trinkle23897** [comment]
> A pr to fix this is also very welcome.

Just an extra argument `deterministic_eval` like SAC, I think you (@ChenDRAG) can do it very quickly, the only thing is to test the performance.

**Comment by ultmaster** [comment]
Yes I can do the PR. I can make it disabled by default, and leave the benchmarking discussions to you guys.

**Comment by Trinkle23897** [comment]
That's cool! Much appreciated! I can do the benchmark.
